generators/csharp/codegen/src/project/CSharpFile.ts

```diff
@@ -10,12 +11,17 @@ export declare namespace CSharpFile {
         clazz: Class | Enum;
         /* Directory of the filepath */
         directory: RelativeFilePath;
+        context: AbstractCsharpGeneratorContext<any>;
     }
 }
 
 export class CSharpFile extends File {
-    constructor({ clazz, directory }: CSharpFile.Args) {
-        super(`${clazz.name}.cs`, directory, clazz.toString());
+    constructor({ clazz, directory, context }: CSharpFile.Args) {
```

Comment:
by requiring `context` this makes it pretty hard to unit test `CsharpFile`. Can we insted take in the values separately. So arguments for `namespace`, `baseNamespaces`, etc.

Reply:
yeah will do